File: Client/Src/Data/PlayerTable.cs
using System.Collections.Generic;
using UnityEngine;

public class PlayerTable
{
    private readonly Dictionary<int, PlayerTableData> _tableById = new();

    public void Load()
    {
        PlayerTableData[] entries = DataLoader.LoadTable<PlayerTableData>("Data/PlayerTable");
        foreach (var data in entries)
            _tableById[data.playerId] = data;

        Debug.Log($"[PlayerTable] Loaded: {entries.Length} entries");
    }

    public PlayerTableData Get(int playerId)
    {
        _tableById.TryGetValue(playerId, out var data);
        return data;
    }
}

File: Server/src/Table/PlayerTable.cpp
#include "../../include/Table/PlayerTable.h"
#include <nlohmann/json.hpp>
#include <fstream>
#include <iostream>
using json = nlohmann::json;

void PlayerTable::Load(const std::string &path) {
    std::ifstream file(path);
    if (!file.is_open()) {
        std::cerr << "[PlayerTable] Failed to open: " << path << std::endl;
        return;
    }

    json arr = json::parse(file);
    for (auto &obj: arr) {
        PlayerTableData data;
        data.playerId = obj["playerId"].get<int>();
        data.maxHp = obj["maxHp"].get<int>();
        data.speed = obj["speed"].get<float>();
        data.defense = obj["defense"].get<int>();
        _tableById[data.playerId] = std::move(data);
    }

    std::cout << "[PlayerTable] Loaded: " << _tableById.size() << " entries" << std::endl;
}

const PlayerTableData *PlayerTable::Get(int playerId) const {
    auto it = _tableById.find(playerId);
    return (it != _tableById.end()) ? &it->second : nullptr;
}
